plans/create.pp

```diff
@@ -74,10 +80,38 @@
   Optional[String[1]] $environment               = 'production',
   Optional[Enum['linux', 'windows']] $os_type    = 'linux',
   Optional[Provision::ProviderOptions] $provider_options = undef,
+
+  ##GCP Specific Parameters
+  Optional[String[1]] $project                    = undef,
+  Optional[String[1]] $network                    = 'default',
+  Optional[String[1]] $subnetwork                 = 'default',
+  Optional[String[1]] $subnetwork_project         = undef,
 ) {
   out::message('Starting infrastructure provisioning')
   $tf_dir = "terraform/${provider}/"
 
+  # Validate the provider options for GCP
+  if $provider == 'gcp' {
```

Comment:
Can we do this in the provision::terraform::apply plan to keep provision::create free from provider dependencies?